publish.gradle

```diff
@@ -37,106 +37,84 @@ def getRepositoryPassword() {
+
+                scm {
+                    url = 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
+                    connection = 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
+                    developerConnection = 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
                 }
 
-
-
-
-                pom.project {
-                    name 'gdx-controllerutils'
-                    description 'Game Controller Utilities for libGDX'
-                    url 'https://github.com/MrStahlfelge/gdx-controllerutils'
-
-                    scm {
-                        url 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
-                        connection 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
-                        developerConnection 'scm:git@github.com:MrStahlfelge/gdx-controllerutils.git'
-                    }
-
-                    licenses {
-                        license {
-                            name 'The Apache Software License, Version 2.0'
-                            url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-                            distribution 'repo'
-                        }
+                licenses {
+                    license {
+                        name = 'The Apache Software License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        distribution = 'repo'
                     }
+                }
 
-                    developers {
-                        developer {
-                            id 'MrStahlfelge'
-                            name 'Benjamin Schulte'
-                            email 'lightblocks@golfgl.de'
-                        }
+                developers {
+                    developer {
+                        id = 'MrStahlfelge'
+                        name = 'Benjamin Schulte'
+                        email = 'lightblocks@golfgl.de'
                     }
                 }
             }
         }
     }
+}
 
-    signing {
-        required { isReleaseBuild() && gradle.taskGraph.hasTask("uploadArchives") }
-        sign configurations.archives
-    }
-
-    task libraryJar(type: Jar, dependsOn:classes) {
-        from sourceSets.main.output.classesDir
-        from configurations.custom.collect { it.isDirectory() ? it : zipTree(it) }
-        classifier = 'library'
-    }
-
-    task sourcesJar(type: Jar, dependsOn:classes) {
-        from sourceSets.main.allSource
-        classifier = 'sources'
-    }
-
-    task javadocJar(type: Jar, dependsOn:javadoc) {
-        from javadoc.destinationDir
-        classifier = 'javadoc'
-    }
-
-    artifacts {
-        archives libraryJar
-        archives sourcesJar
-        archives javadocJar
-    }
 
+signing {
+    required { isReleaseBuild() && gradle.taskGraph.hasTask("uploadArchives") }
+    sign publishing.publications.maven
```

Comment:
Something does not work yet with signing, it is producing a NullPointerException on build

Reply:
What did you run to test it? I ran `gradlew clean publish -PLOCAL=true` now and it seems fine